configure.ac

```diff
@@ -24,17 +25,30 @@ LT_PREREQ([2.4])
 LT_INIT()
 
 # Process user optons.
-
-# Does the user want to turn off all attempts to set fortran flags?
-# This is necessary to handle situations where the configure-provided
-# flags will be wrong for a particular install. With this setting, the
-# user is in complete control of Fortran flags.
-AC_MSG_CHECKING([whether we should attempt to set Fortran flags for the user])
-AC_ARG_ENABLE([fortran-flag-setting],
-  [AS_HELP_STRING([--disable-fortran-flag-setting],
-    [Disable any attempts by configure to set fortran flags. With this option, only user-provided flags (set in FCFLAGS) will be used compiling Fortran code.])])
-test "x$enable_fortran_flag_setting" = xno || enable_fortran_flag_setting=yes
-AC_MSG_RESULT([$enable_fortran_flag_setting])
+AC_ARG_ENABLE([mixed-mode],
+  [AS_HELP_STRING([--enable-mixed-mode],
+    [Build using mixed mode.  Enables both 64-bit and 32-bit reals in Fortran. This option will be ignored if --disable-fortran-flag-setting is also given.])])
+AS_IF([test ${enable_mixed_mode:-no} = no],
+  [enable_mixed_mode=no],
+  [enable_mixed_mode=yes])
+AC_ARG_WITH([mpi],
```

Comment:
This is not the correct usage of the --with- feature for autoconf.

Use --with-XXX when XXX is a library that can optionally be used, and you may want to specify an alternate location.

This is not the case with MPI. Firstly, isn't MPI always required? Secondly, using MPI involves more than just the lib and include directories, additionally the compiler is different. This was a dumb decision by MPI but we are stuck with it.

So when the user says CC=mpicc ./configure --with-mpi=/some/other/location you will not be able to meet that request, since you don't control mpicc and where it looks for MPI.

What are you trying to achieve with --with-mpi here? Is it possible to build this code without MPI? If so, we can check for an MPI function in configure.ac, and set a flag accordingly. The user does not need to do anything except set FC/CC to MPI compilers.

Reply:
The library can be built without MPI, and is done so in some of the GFDL idealized test cases (used internally by some GFDL scientists).  It is also possible to build with MPI without using the MPI compiler wrappers (`mpicc`, `mpif90`, etc.).  In fact, that was the standard way we built the library using our older build process (using `mkmf`).  At this time, the only valid responses for `--with-mpi` is `yes` or `no`.  I probably should have included a check for anything other than those two responses.  It may be possible in a future commit to make use of a path, but we will not support this usage ATT.

Reply:
OK, then you should use --enable and not --with for this.